File: apps/gig-crawler-2-python-backup/src/models/scraped_content.py
"""Scraped content models for Pass 2."""
from typing import Optional
from pydantic import BaseModel


class ScrapedContent(BaseModel):
    """Scraped content from trafilatura."""
    url: str
    text: Optional[str] = None
    raw_html: Optional[str] = None
    success: bool = True
    error: Optional[str] = None

File: apps/gig-crawler-2/src/models/scrapedContent.ts
import { z } from "zod";

export const ScrapedContentSchema = z.object({
  url: z.string(),
  text: z.string().optional(),
  rawHtml: z.string().optional(),
  success: z.boolean(),
  error: z.string().optional(),
});

export type ScrapedContent = z.infer<typeof ScrapedContentSchema>;
